Tests failing on master

@tswast do you recognize these? https://travis-ci.org/pydata/pandas-gbq/jobs/364877445

I can't immediately see how 8f49ec3f134cc0a85f541116c42647c515fdc7e6 could have caused those

```
self = <MagicMock name='mock.query()' spec='QueryJob' id='139858594198360'>
name = 'job_id'
    def __getattr__(self, name):
        if name in ('_mock_methods', '_mock_unsafe'):
            raise AttributeError(name)
        elif self._mock_methods is not None:
            if name not in self._mock_methods or name in _all_magics:
>               raise AttributeError("Mock object has no attribute %r" % name)
E               AttributeError: Mock object has no attribute 'job_id'
.nox/test35/lib/python3.5/site-packages/mock/mock.py:698: AttributeError
```

Odd. I would have thought the QueryJob spec would have worked here. I can't think of any changes in google-cloud-bigquery that would have caused it.

--- a/.travis.yml
+++ b/.travis.yml
@@ -37,9 +37,6 @@ install:
       conda create -q -n test-environment python=$PYTHON ;
       source activate test-environment ;
       conda install -q pandas=$PANDAS;
-      pip install --upgrade setuptools
-      pip install --upgrade pip
-      pip install coverage pytest pytest-cov flake8 codecov ;
       conda install -q --file "$REQ.conda";
       conda list ;
       python setup.py install ;

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,6 +17,7 @@ install:
   # https://github.com/pre-commit/pre-commit/commit/e3ab8902692e896da9ded42bd4d76ea4e1de359d
   - pyenv install -s $PYENV_VERSION
   - pyenv global system $PYENV_VERSION
+  - pip install --upgrade setuptools
   - REQ="ci/requirements-${PYTHON}-${PANDAS}" ;
     if [ -f "$REQ.pip" ]; then
       pip install --upgrade nox-automation ;
@@ -33,6 +34,7 @@ install:
       conda create -q -n test-environment python=$PYTHON ;
       source activate test-environment ;
       conda install -q pandas=$PANDAS;
+      pip install --upgrade setuptools
       pip install coverage pytest pytest-cov flake8 codecov ;
       conda install -q --file "$REQ.conda";
       conda list ;

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,7 @@ install:
     if [ -f "$REQ.pip" ]; then
       pip install --upgrade nox-automation ;
     else
-      wget http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+      wget http://repo.continuum.io/miniconda/Miniconda3-4.3.30-Linux-x86_64.sh -O miniconda.sh;
       bash miniconda.sh -b -p $HOME/miniconda ;
       export PATH="$HOME/miniconda/bin:$PATH" ;
       hash -r ;

--- a/.travis.yml
+++ b/.travis.yml
@@ -48,7 +48,6 @@ script:
   - if [[ $PYTHON == '3.6' ]] && [[ "$PANDAS" == "MASTER" ]]; then nox -s test36master ; fi
   - REQ="ci/requirements-${PYTHON}-${PANDAS}" ;
     if [ -f "$REQ.conda" ]; then
-      pip install pytest ;
       pytest -v tests ;
     fi
   - if [[ $COVERAGE == 'true' ]]; then nox -s cover ; fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,10 @@ install:
   # https://github.com/pre-commit/pre-commit/commit/e3ab8902692e896da9ded42bd4d76ea4e1de359d
   - pyenv install -s $PYENV_VERSION
   - pyenv global system $PYENV_VERSION
+  # Upgrade setuptools and pip to work around
+  # https://github.com/pypa/setuptools/issues/885
   - pip install --upgrade setuptools
+  - pip install --upgrade pip
   - REQ="ci/requirements-${PYTHON}-${PANDAS}" ;
     if [ -f "$REQ.pip" ]; then
       pip install --upgrade nox-automation ;
@@ -35,6 +38,7 @@ install:
       source activate test-environment ;
       conda install -q pandas=$PANDAS;
       pip install --upgrade setuptools
+      pip install --upgrade pip
       pip install coverage pytest pytest-cov flake8 codecov ;
       conda install -q --file "$REQ.conda";
       conda list ;

--- a/.travis.yml
+++ b/.travis.yml
@@ -36,6 +36,7 @@ install:
       conda info -a ;
       conda create -q -n test-environment python=$PYTHON ;
       source activate test-environment ;
+      conda install -q setuptools ;
       conda install -q pandas=$PANDAS;
       conda install -q --file "$REQ.conda";
       conda list ;

--- a/ci/requirements-3.6-0.20.1.conda
+++ b/ci/requirements-3.6-0.20.1.conda
@@ -1,7 +1,6 @@
 google-auth
 google-auth-oauthlib
-google-api-core
-google-cloud-bigquery
+google-cloud-bigquery==0.32.0
 pytest
 pytest-cov
 codecov

--- a/ci/requirements-3.6-0.20.1.conda
+++ b/ci/requirements-3.6-0.20.1.conda
@@ -1,5 +1,6 @@
 google-auth
 google-auth-oauthlib
+google-api-core
 google-cloud-bigquery
 pytest
 pytest-cov

--- a/ci/requirements-3.6-0.20.1.conda
+++ b/ci/requirements-3.6-0.20.1.conda
@@ -1,3 +1,8 @@
 google-auth
 google-auth-oauthlib
 google-cloud-bigquery
+pytest
+pytest-cov
+codecov
+coverage
+flake8

--- a/tests/unit/test_gbq.py
+++ b/tests/unit/test_gbq.py
@@ -20,6 +20,7 @@ def mock_bigquery_client(monkeypatch):
     mock_client = mock.create_autospec(google.cloud.bigquery.Client)
     # Mock out SELECT 1 query results.
     mock_query = mock.create_autospec(google.cloud.bigquery.QueryJob)
+    mock_query.job_id = 'some-random-id'
     mock_query.state = 'DONE'
     mock_rows = mock.create_autospec(
         google.cloud.bigquery.table.RowIterator)

